### ISSUE TITLE
Delete selected item when pressing backspace button

### ISSUE THREAD

**Issue by carstenhag** [openstreetmap/iD#6862]
I'm using a Mac, and a novice mac user might not know that in order to press "delete", Fn+Backspace needs to be pressed.

Even users of other OS might expect backspace to delete things.

**Comment by leifr-unc** [comment]
This might lead to accidental deletions when users are trying to type.  I personally find the current shortcuts discoverable enough.

**Comment by bhousel** [comment]
I thought iD already treated all the combinations of delete and backspace the same?  Is there a difference?

**Comment by BjornRasmussen** [comment]
(reposting) Just backspace might lead to accidental deletions when users are trying to type.  I personally find the current shortcuts discoverable enough for most people.  I suppose that some people could find this confusing though.

**Comment by quincylvania** [comment]
@carstenhag Are you using a non-Apple keyboard? I'm not aware of modern Apple keyboards with a Backspace key.

@BjornRasmussen I don't think this about single-key deletion, but about needing an extra modifier key (<kbd>fn</kbd>) to enable one of the keys, leading to three keys need for deletion.

**Comment by carstenhag** [comment]
![IMG_20190917_220705](https://user-images.githubusercontent.com/3776997/65075585-853a7280-d997-11e9-812c-e5f58c49f3fd.jpg)

It's a backspace, isn't it? :D (MBP late 2015)

**Comment by quincylvania** [comment]
@carstenhag Interesting! On the US keyboards it just says "delete" without an arrow. 😅

I assume it works the same. You're saying you need to hit <kbd>fn</kbd> to delete a feature with it in iD?

**Comment by carstenhag** [comment]
Oh, I wasn't aware of the difference. It's a german layout. Exactly, fn+backspace is a delete.

**Comment by carstenhag** [comment]
Just to be clear, a somewhat experienced Mac user knows this shortcut. Some of my colleagues that use this same MacBook since 3 years don't though.

**Comment by bhousel** [comment]
> Oh, I wasn't aware of the difference. It's a german layout. Exactly, fn+backspace deletes things in ID

1. Does <kbd>Cmd⌘</kbd> + <kbd>that key</kbd> delete things too?  (without <kbd>fn</kbd>)  
1. Does the popup documentation for the delete operation make sense?  :

<img width="283" alt="Screenshot 2019-09-17 17 54 00" src="https://user-images.githubusercontent.com/38784/65082555-3f1fe780-d974-11e9-9365-bb65263a2ac0.png">

and on the <kbd>?</kbd> screen:

<img width="304" alt="Screenshot 2019-09-17 18 05 31" src="https://user-images.githubusercontent.com/38784/65083164-cde13400-d975-11e9-85a4-8853a86e9921.png">


Looking at the code, we try to cover all these cases and have them work the same, but we document the first one in the help screens..  (I suspect you're hitting the third case below when you use <kbd>fn</kbd> modifier to turn the mac backspace ⌫ into a mac delete ⌦.)
https://github.com/openstreetmap/iD/blob/4dd6d3f02622c872f3ddad4583e1a25ca9d5616a/modules/operations/delete.js#L150

**Comment by bhousel** [comment]
For the curious #1698 contains the thread about why the shortcuts are the way they are.

tl;dr - people complained that "iD users will delete too many things by accident" so we made the shortcut slightly trickier, despite there being no evidence to support this claim.

**Comment by quincylvania** [comment]
> tl;dr - people complained that "iD users will delete too many things by accident" so we made the shortcut slightly trickier, despite there being no evidence to support this claim.

I'd be up for reevaluating this. "Why doesn't the delete button work" seems like it'd be a real annoyance. We also have guardrails like preventing deletion of offscreen features and those linked to Wikidata, as well as showing a deletion message at the bottom of the screen. It'd be hard to delete something by accident and not notice.

Also, as I mentioned in #6756 and #6475, it'd be cool to have a "Power Delete" shortcut to bypass downgrade steps. To do this now we'd need to add an additional modifier key, but it'd be simpler to map Downgrade to <kbd>Delete</kbd> and Power Delete to <kbd>Cmd</kbd> + <kbd>Delete</kbd>.

**Comment by tordans** [comment]
> * Does Cmd⌘ + that key delete things too?  (without fn)
> * Does the popup documentation for the delete operation make sense?  :

Yes to both for my German keyboard. It looks like fn+backspace == cmd+backspace.
[Tested here](https://preview.ideditor.com/master/#background=Berlin-2019&disable_features=boundaries&locale=de&map=20.48/52.47761/13.44944).

> … It'd be hard to delete something by accident and not notice.

A compromise UI for this could be:
- just backspace shows a "are you sure to delete" dialogue
- cmd+backspace skips this dialogue
- there is a "dont show this message" checkbox on the dialogue (locale storage) that hides it

Personally I thinks the current behaviour is fine. There is always the right mouse button _and_ the new toolbar icon to delete.

> … "Power Delete" …

Personally I did not feel like I need this, yet.

**Comment by quincylvania** [comment]
For #7682 I mapped Backspace to the Downgrade operation and made Cmd+Backspace delete immediately. When the selection is downgraded all the way, I think we could do @tordans's suggestion:

> - just backspace shows a "are you sure to delete" dialogue
> - cmd+backspace skips this dialogue

Not sure about this part since mappers can just use the two-key shortcut:

> - there is a "dont show this message" checkbox on the dialogue (locale storage) that hides it
